src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IDynamicIndexerAccessExpression.cs

```diff
@@ -377,5 +377,563 @@ void M(C c, dynamic d)
 
             VerifyOperationTreeAndDiagnosticsForTest<ElementAccessExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void DynamicIndexerAccess_ControlFlowInReceiver()
+        {
+            string source = @"
+class C
+{
+    void M(C c1, C c2, dynamic d, dynamic p)
+    /*<bind>*/{
+        p = (c1 ?? c2)[d];
```

Comment:
>p = (c1 ?? c2)[d]; [](start = 8, length = 18)

It would be good to have a test covering scenario without any control flow in this construct. It will help us to confirm that there is no unnecessary stack spilling is happening.

Reply:
Sure.

---
In reply to: [186830815](https://github.com/dotnet/roslyn/pull/26690#discussion_r186830815) [](ancestors = 186830815)

Reply:
Added `DynamicIndexerAccess_NoControlFlow`

---
In reply to: [186831047](https://github.com/dotnet/roslyn/pull/26690#discussion_r186831047) [](ancestors = 186831047,186830815)